src/gui/stac/qgsstacsourceselect.cpp

```diff
@@ -297,43 +299,53 @@ void QgsStacSourceSelect::onStacObjectRequestFinished( int requestId, QString er
       // collections endpoint should have a "data" relation according to spec but some servers don't
       // so let's be less strict and only check the href
       if ( l.href().endsWith( "/collections" ) )
-        mCollectionsUrl = l.href();
+        collectionsUrl = l.href();
       else if ( l.relation() == "search" )
         mSearchUrl = l.href();
 
-      if ( !mCollectionsUrl.isEmpty() && !mSearchUrl.isEmpty() )
+      if ( !collectionsUrl.isEmpty() && !mSearchUrl.isEmpty() )
         break;
     }
   }
 
-  if ( mCollectionsUrl.isEmpty() || mSearchUrl.isEmpty() )
+  if ( collectionsUrl.isEmpty() || mSearchUrl.isEmpty() )
   {
     mStatusLabel->setText( tr( "Server does not support STAC search API" ) );
   }
   else
   {
-    mStatusLabel->setText( tr( "Fetching Collections…" ) );
-    mStac->cancelPendingAsyncRequests();
-    mStac->fetchCollectionsAsync( mCollectionsUrl );
+    mStac->fetchCollectionsAsync( collectionsUrl );
+    mParametersDialog->clearCollections();
+    mStatusLabel->setText( tr( "Fetching collections…" ) );
   }
 }
 
 void QgsStacSourceSelect::onCollectionsRequestFinished( int requestId, QString error )
 {
   QgsDebugMsgLevel( QStringLiteral( "Finished collections request %1" ).arg( requestId ), 2 );
-  QgsStacCollections *cols = mStac->takeCollections( requestId );
+  std::unique_ptr<QgsStacCollections> cols( mStac->takeCollections( requestId ) );
 
   if ( !cols )
   {
     mStatusLabel->setText( error );
-    mParametersDialog->setCollections( {} );
-    updateFilterPreview();
     return;
   }
 
   const QVector<QgsStacCollection *> vcols = cols->takeCollections();
-  mParametersDialog->setCollections( vcols );
+  mParametersDialog->appendCollections( vcols );
   mItemsModel->setCollections( vcols );
+
+  // Let's try to grab the first 5 pages of /collections endpoint before searching
+  // In most cases all collections will be returned in the first page but some servers are weird
+  if ( mCollectionsPageCounter < 5 && !cols->nextUrl().isEmpty() )
+  {
+    ++mCollectionsPageCounter;
```

Comment:
it seems the page counter never gets reset? it probably should be when connecting...

Reply:
good catch, fixed!